jakarta-eclipse/src/org/jakarta/lsp4e/JakartaLanguageClient.java

```diff
@@ -57,22 +65,49 @@ public boolean isCanceled() {
     public CompletableFuture<List<PublishDiagnosticsParams>> getJavaDiagnostics(
             JakartaDiagnosticsParams javaParams) {
         Activator.log(new Status(IStatus.INFO, "diagnostic request received", "diagnostic request receieved"));
-        
         // creating a test diagnostic
         return CompletableFutures.computeAsync((cancelChecker) -> {
             IProgressMonitor monitor = getProgressMonitor(cancelChecker);
                 List<PublishDiagnosticsParams> publishDiagnostics = new ArrayList<PublishDiagnosticsParams>();
+                
+                List<Diagnostic> diagnostics = new ArrayList<>();
                 List<String> uris = javaParams.getUris();
                 for (String uri : uris) {
-                    List<Diagnostic> diagnostics = new ArrayList<>();
-                    Range diagRange = new Range(new Position(2, 4), new Position(2, 10));
-                    diagnostics.add(new Diagnostic(diagRange, "testDiagnostic"));
+                	
+                	URI u = JDTUtils.toURI(uri);
+                	ICompilationUnit unit = JDTUtils.resolveCompilationUnit(u);
+                	if (unit != null) {
+                		System.out.println("--class name: " + unit.getElementName());
+                    	IType[] alltypes;
+    					try {
+    						alltypes = unit.getAllTypes();
+    						for (IType type: alltypes)  {
+    	                		IMethod[] methods = type.getMethods();
+    	                		for (IMethod method : methods) {
+    								System.out.println("--Method name: "+ method.getElementName());
+    								System.out.println("--Signature: "+ method.getSignature());
+    								System.out.println("--Return Type: "+ method.getReturnType());
+    								System.out.println("--source: "+ method.getSource());
+    								System.out.println("--to string: "+ method.toString());
+    								System.out.println("--new: "+ method.getPath().toString());
```

Comment:
we can get rid of these System.out.println statements, if we want them for future reference, use a comment or log them with `Activator.log`